Recast/Source/RecastAlloc.cpp

```diff
@@ -77,6 +78,7 @@ void rcIntArray::doResize(int n)
 	if (!m_cap) m_cap = n;
 	while (m_cap < n) m_cap *= 2;
 	int* newData = (int*)rcAlloc(m_cap*sizeof(int), RC_ALLOC_TEMP);
+	rcAssert(newData);
 	if (m_size && newData) memcpy(newData, m_data, m_size*sizeof(int));
```

Comment:
It will still crash later when the array is used, so that check could probably be eliminated.
I don't really see what we can do here - the whole problem is that, to detect that it has happened, we need to return an error code, and that needs to be bubbled up everywhere.
What we could do instead is use `std::terminate` or `std::abort` from inside this function, but I don't really think there's much point in it - I think it is very unlikely someone is meshing in environments where allocations in `rcIntArray` will fail (as said previously the allocations by it are very small usually).